django/contrib/auth/base_user.py

```diff
@@ -77,6 +73,9 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
         return self.get_username()
 
+    def clean(self):
+        setattr(self, self.USERNAME_FIELD, self.normalize_username(self.get_username()))
```

Comment:
No, providing `USERNAME_FIELD` is a requirement for subclassing `AbstractBaseUser`: https://docs.djangoproject.com/en/1.9/topics/auth/customizing/#django.contrib.auth.models.CustomUser